solver/llbsolver/solver.go

```diff
@@ -119,12 +132,22 @@ func New(opt Opt) (*Solver, error) {
 	}
 	s.sysSampler = sampler
 
+	refIDStore, err := solver.NewRefIDStore(opt.RootDir)
+	if err != nil {
+		return nil, err
+	}
+
+	sources := worker.NewCombinedResultSource(
```

Comment:
Here's a new abstraction that allows the simple solver to load from a set of result sources. The inline cache remote source is specified here.